src/middleware.js

```diff
@@ -112,6 +112,82 @@ function errorHandler(err, req, res, next) {
   }
 }
 
+function setHttpCacheHeaders(cacheObj) {
+  /*
+ Level 1 - No Caching at all
+ Level 2 - Some caching, but for a small time window, 5 minutes
+ Level 3 - Regular Caching, store them for a year
+ Level 4 - Extreme Caching, store them forever
+ */
+
+  const options = cacheObj.options;
+  const levelValue = cacheObj.level;
+  const res = cacheObj.res;
+
+  // Enum the levels for readability
+  const levels = { LEVEL1: 1, LEVEL2: 2, LEVEL3: 3, LEVEL4: 4 };
+
+  if (level === null && options === null) {
+    next(
+      createError(
+        500,
+        'You passed in arguments that are of type NULL. Make sure the objects exist before calling the function.'
+      )
+    );
+  }
+
+  //If we want to explicitly set a bunch of headers at once, without using the default levels.
+  // Passed as a JSON Object with the following structure: {Cache-Control-Option : value }
+  if (level === null || level === 0) {
+    try {
+      for (var key in options) {
+        if (options.hasOwnProperty(key)) {
+          r.set('${key}', '${options[key]}');
+        }
+      }
+    } catch (error) {
+      next(
+        createError(
+          500,
+          "Unexpected Error: ${error}, check to see if you've made a mistake when defining options."
+        )
+      );
+    }
+  } else {
+    switch (levelValue) {
+      case LEVEL1:
+        res.set('Cache-Control', 'public, max-age=0');
```

Comment:
Is there any way this could (_should?_) allow the dev to specify their own `max-age`? Kind of like how you can pass options in your example?

I'm asking mostly because I'm not sure if this is functionality is something that **should** be added in, or if we should only allow what is defined, no more, no less.

Reply:
Also the jump from 300 to 2592000 is pretty big, should it be? (genuine question, I'm not sure)

Reply:
That's a good question, I originally designed it so that we could set pre-defined limits on how long we could cache our data. 

However you make a good point, cause maybe we might need something to be cached for a specific time. 

I'll add in an option for that, just in case, Thanks 😄

Reply:
Np!